chasm/lib/activity/activity.go

```diff
@@ -679,3 +696,103 @@ func (a *Activity) buildPollActivityExecutionResponse(
 		FrontendResponse: response,
 	}, nil
 }
+
+// recordOnAttemptedMetrics records metrics for attempted activities, including retries and originating from any
+// terminal state transitions.
+func (a *Activity) recordOnAttemptedMetrics(
+	startedTime time.Time,
+	namespaceName string,
+	metricsHandler metrics.Handler,
+	breakdownMetricsByTaskQueue dynamicconfig.BoolPropertyFnWithTaskQueueFilter,
+	operationTag string,
+	timeoutType enumspb.TimeoutType,
+) {
+	taskQueueFamily := a.GetTaskQueue().GetName()
+
+	handler := metrics.GetPerTaskQueueFamilyScope(
```

Comment:
I would suggest passing in a fully baked metrics handler into the application logic.

You can build the metrics handler in the API handlers or task executors if needed, it will save you from carrying all of these parameters around.